File: output/pascal/demo_rozszerzenia.pas
program DemoRozszerzenia ;

// Stare: const
const
    MAX = 100 ;
    ETYKIETA = "demo" ;

// Stare + nowe: var (int, bool, string, tablica, real, char, record)
var
    i : integer ;
    suma : integer ;
    ok : boolean ;
    tekst : string ;
    waga : real ;
    sr : real ;
    znak : char ;
    liczby : array ( 1 .. 3 ) of integer ;
    punkt : record begin
        x : integer ;
        y : integer ;
    end ;

// Nowe: BYREF (var)
procedure Zwieksz ( var v : integer ) ;
begin
    v := v + 1 ;
end ;

// Stare: function + nowe: real
function Srednia ( a : integer ; b : integer ) : real ;
begin
    exit  ( a + b ) : real / 2.0 ;
end ;

begin
    // Stare: assign, wyrazenia, bool
    suma := 0 ;
    ok := true ;
    tekst := ETYKIETA ;
    znak := 'X' ;
    waga := 3.5 ;

    // Nowe: pola rekordu (.)
    punkt . x := 10 ;
    punkt . y := 20 ;

    // Stare: for + tablica
    for i := 1 to 3 do
        liczby ( i ) := i * 10 ;
        suma := suma + i * 10 ;

    // Nowe: CAST () — int na real przed dzieleniem
    waga :=  ( suma ) : real / 2.0 ;

    // Stare: while
    while ok and suma < MAX do
        Zwieksz ( suma ) ;

    // Stare: if
    if suma >= 50 then ok := true else ok := false ;

    // Stare: repeat-until
    repeat
        suma := suma - 1 ;
    until suma = 50 ;

    // Stare: case
    case punkt . x of
        10 then punkt . y := punkt . y + 1 ;
        20 then punkt . y := punkt . y - 1 ;
        else punkt . y := 0 ;
    end ;

    // Stare: print (int, string, bool, real, char, record fields)
    writeln ( tekst ) ;
    writeln ( suma ) ;
    writeln ( ok ) ;
    writeln ( waga ) ;
    writeln ( znak ) ;
    writeln ( punkt . x ) ;
    writeln ( punkt . y ) ;
    sr := Srednia ( punkt . x , punkt . y ) ;
    writeln ( sr ) ;
end .

File: output/c/demo_rozszerzenia.c
#include <stdio.h>

/* Program: DemoRozszerzenia */
typedef enum { Czerwony = 0, Zielony = 1, Niebieski = 2 } Kolor;
typedef unsigned int Set_Kolor;

typedef struct {
    int x;
    int y;
} Record_1;

#define MAX 100
#define ETYKIETA "demo"

void Zwieksz(int *v) {
    (*v) = ((*v) + 1);
}
double Srednia(int a, int b) {
    return (((double)(a + b)) / 2.0);
}

int main(void) {
    int i;
    int suma;
    int ok;
    char* tekst;
    double waga;
    double sr;
    char znak;
    int liczby[3];
    Record_1 punkt;
    Set_Kolor barwy;
    Kolor aktywny;
    
    suma = 0;
    ok = 1;
    tekst = ETYKIETA;
    znak = 'X';
    waga = 3.5;
    punkt.x = 10;
    punkt.y = 20;
    for (i = 1; i <= 3; i++) {
        liczby[(i - 1)] = (i * 10);
    }
    suma = (suma + (i * 10));
    waga = (((double)suma) / 2.0);
    while ((ok && (suma < MAX))) {
        Zwieksz(&suma);
    }
    if ((suma >= 50)) {
        ok = 1;
    } else {
        ok = 0;
    }
    do {
        suma = (suma - 1);
    } while (!((suma == 50)));
    barwy = (1u << Czerwony) | (1u << Niebieski);
    barwy = (barwy | (1u << Zielony));
    aktywny = Czerwony;
    if (((barwy & (1u << Czerwony)) != 0)) {
        ok = 1;
    } else {
        ok = 0;
    }
    switch (aktywny) {
    case Czerwony:
        suma = (suma + 1);
        break;
    case Zielony:
        suma = (suma + 2);
        break;
    default:
        suma = suma;
        break;
    }
    switch (punkt.x) {
    case 10:
        punkt.y = (punkt.y + 1);
        break;
    case 20:
        punkt.y = (punkt.y - 1);
        break;
    default:
        punkt.y = 0;
        break;
    }
    printf("%s\n", tekst);
    printf("%d\n", suma);
    printf("%d\n", ok);
    printf("%lf\n", waga);
    printf("%c\n", znak);
    printf("%d\n", punkt.x);
    printf("%d\n", punkt.y);
    printf("%u\n", barwy);
    printf("%d\n", aktywny);
    sr = Srednia(punkt.x, punkt.y);
    printf("%lf\n", sr);
    return 0;
}
